File: frontend/lib/ticket.js
/**
 * lib/ticket.js
 * -------------
 * Deterministic, human-friendly tracking ticket derived from an issue's UUID.
 * The same issue id always yields the same ticket, so it can be shown on the
 * success popup AND later in My History for future reference.
 */

export function ticketNumber(id) {
  if (!id) return "FMS-UNKNOWN";
  const hex = id.replace(/-/g, "").slice(0, 8).toUpperCase();
  return `FMS-${hex}`;
}

File: mobile/lib/domain/ticket.dart
/// Port of lib/ticket.js — deterministic, human-friendly tracking ticket
/// derived from an issue's UUID. Same id → same ticket, always.
String ticketNumber(String? id) {
  if (id == null || id.isEmpty) return 'FMS-UNKNOWN';
  final hex = id.replaceAll('-', '');
  final head = hex.length >= 8 ? hex.substring(0, 8) : hex;
  return 'FMS-${head.toUpperCase()}';
}
